odoo/models.py

```diff
@@ -3641,6 +3716,15 @@ def _setup_base(self):
         elif 'x_active' in cls._fields:
             cls._active_name = 'x_active'
 
+        # 7. determine sql_objects
+        sql_objects = {}
+        for klass in reversed(cls._model_classes):
+            # this condition is an optimization of is_definition_class(klass)
+            if isinstance(klass, MetaModel):
+                sql_objects.update(klass._local_sql_objects)
+        sql_objects.update(cls._local_sql_objects)
+        cls._sql_objects = frozendict(sql_objects)
```

Comment:
To be discussed: since `_sql_object` is only used when `install`/`update` maybe we can cleanup it after registry fully loaded.

or maybe we can use `@property` for it?

Reply:
It is used in `_sql_error_to_message` to resolve the message of errors. We need to keep it at runtime.